File: step2/lect1/bubblesort/BubbleSort.java
package bubblesort;

class BubbleSort {
    public static void main(String[] args) {
        int[] arr = new int[] { 9, 8, 7, 6, 3, 4, 5, 1 };

        sort(arr);
    }

    static void sort(int[] arr) {

        while (true) {
            boolean swapped = false;

            for (int j = 0; j < arr.length - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;

                    swapped = true;
                }
            }

            print(arr);

            if (!swapped)
                break;
        }
    }

    static void print(int[] arr) {
        for (int i : arr) {
            System.out.print(i + " ");
        }

        System.err.println();
    }
}

File: step2/lect1/bubblesort/bubblesort.go
package main

import "fmt"

func main() {
	var arr [9]int = [9]int{9, 8, 7, 6, 2, 3, 4, 1, 5}

	for true {
		swapped := false

		for i := 0; i < len(arr)-1; i++ {
			if arr[i] > arr[i+1] {
				temp := arr[i]

				arr[i] = arr[i+1]
				arr[i+1] = temp

				swapped = true
			}
		}

		if !swapped {
			break
		}

		fmt.Println(arr)
	}
}
